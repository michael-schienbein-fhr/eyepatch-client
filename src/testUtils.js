import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  photo_url: null,
};

const demoQueueVideo = {
  username: 'testuser2',
  type: 'video',
  action: 'add',
  text: '"testVideoTitle" added to queue for user: "testuser2".',
  videoId: 'testVideoId',
  title: 'testVideoTitle',
  description: 'testDescription',
  thumbnail: 'testThumbnail'
}
const demoSearchVideo = { "kind": "youtube#searchResult", "etag": "3_5Vn-wWRzIN5QC7aOEWvZsmMwI", "id": { "kind": "youtube#video", "videoId": "GjbibR6DV1s" }, "snippet": { "publishedAt": "2015-08-11T17:03:57Z", "channelId": "UCi_ZH3bYHidyHsETH5hbTFQ", "title": "Dusky - Skin Deep (Official Video)", "description": "Dusky's EP 'Ordinary World' is out now. iTunes:http://po.st/DuskyowEP Spotify: http://po.st/DuskyOWSp Beatport: http://po.st/OrdinaryWorldEP \"This EP was all ...", "thumbnails": { "default": { "url": "https://i.ytimg.com/vi/GjbibR6DV1s/default.jpg", "width": 120, "height": 90 }, "medium": { "url": "https://i.ytimg.com/vi/GjbibR6DV1s/mqdefault.jpg", "width": 320, "height": 180 }, "high": { "url": "https://i.ytimg.com/vi/GjbibR6DV1s/hqdefault.jpg", "width": 480, "height": 360 } }, "channelTitle": "DuskyMusic", "liveBroadcastContent": "none", "publishTime": "2015-08-11T17:03:57Z" } }

const testRoom = {
  "id": 1,
  "roomOwner": "testadmin",
  "roomName": "testadmin",
  "roomMembers": null,
  "videoQueue": null,
  "hasPass": true,
  "created_at": "2021-10-22T10:34:59.443Z"
}
const UserProvider =
  ({ children, currentUser = demoUser }) => (
    <UserContext.Provider value={{ currentUser }}>
      {children}
    </UserContext.Provider>
  );

export { UserProvider, demoUser, demoQueueVideo, demoSearchVideo, testRoom };

