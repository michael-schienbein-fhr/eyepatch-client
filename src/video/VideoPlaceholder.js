const VideoPlaceholder = () => {
  return (
    <div className='placeholder'>
      <div className='text'>
        <p>Start by searching for a video and adding it to the queue.</p>
        <p>You can then click the video to begin watching.</p>
      </div>
    </div>
  )
}

export default VideoPlaceholder;