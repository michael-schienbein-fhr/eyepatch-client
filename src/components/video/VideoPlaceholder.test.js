import { render } from "@testing-library/react";
import userEvent from '@testing-library/user-event'
import { UserProvider, demoQueueVideo, demoSearchVideo } from "../../testUtils";
import VideoPlaceholder from "./VideoPlaceholder";

// const BASE_URL = process.env.REACT_APP_BASE_URL || "http://localhost:3001";
// const nock = require('nock')



it("matches snapshot", function () {
  window.HTMLElement.prototype.scrollIntoView = function () { };
  const { asFragment } = render(
    <UserProvider>
      <VideoPlaceholder
      />
    </UserProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
