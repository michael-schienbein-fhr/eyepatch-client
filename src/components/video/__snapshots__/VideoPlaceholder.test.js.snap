// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="placeholder"
  >
    <div
      class="text"
    >
      <p>
        Start by searching for a video and adding it to the queue.
      </p>
      <p>
        You can then click the video to begin watching.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
